### ISSUE TITLE
Changing class weights in yolov11n 

### ISSUE THREAD

**Issue by shrutichakraborty** [ultralytics/ultralytics#17397]
### Search before asking

- [X] I have searched the Ultralytics YOLO [issues](https://github.com/ultralytics/ultralytics/issues) and [discussions](https://github.com/ultralytics/ultralytics/discussions) and found no similar questions.

### Question

Hello! I am training a yolov11n network on my custom dataset with only 1 single class. The confusion matrix I obtain shows that all the background images are incorrectly classified to contain detections. This is potentially due to a class imbalance. In my training set I have 400 imgaes and 58 background images, and in the validation set I have 216 images, and 18 background images. I would like to know how I can update the weights attributed to the classes during training? 

Thanks!

**Comment by UltralyticsAssistant** [comment]
👋 Hello @shrutichakraborty, thank you for reaching out to the Ultralytics community 🚀! It sounds like you're encountering issues with class imbalance, which can indeed impact the performance of your model. 

To address this, we encourage you to review the [Docs](https://docs.ultralytics.com), specifically the [model training tips](https://docs.ultralytics.com/guides/model-training-tips/) that might help optimize your training process. 

If you suspect this is a bug 🐛, please help us by providing a [minimum reproducible example](https://docs.ultralytics.com/help/minimum_reproducible_example/). This would greatly assist us in diagnosing the problem.

In the meantime, ensure that you have updated to the latest version of `ultralytics` using the following command, as improvements and bug fixes are continually being made:

```bash
pip install -U ultralytics
```

Additionally, consider joining our Ultralytics community for more interactive support:
- For real-time discussions, jump into our [Discord](https://ultralytics.com/discord) 🎧.
- Engage in deeper discussions on [Discourse](https://community.ultralytics.com).
- Or share insights and seek advice on our [Subreddit](https://reddit.com/r/ultralytics).

Your issue will soon receive more personalized attention as an Ultralytics engineer reviews it. Thank you for your patience and for contributing to our community! 😊

## Environments

You can also run YOLO in any of the following verified environments with all dependencies preinstalled:
- **Notebooks** with free GPU: <a href="https://console.paperspace.com/github/ultralytics/ultralytics"><img src="https://assets.paperspace.io/img/gradient-badge.svg" alt="Run on Gradient"/></a> <a href="https://colab.research.google.com/github/ultralytics/ultralytics/blob/main/examples/tutorial.ipynb"><img src="https://colab.research.google.com/assets/colab-badge.svg" alt="Open In Colab"></a> <a href="https://www.kaggle.com/models/ultralytics/yolo11"><img src="https://kaggle.com/static/images/open-in-kaggle.svg" alt="Open In Kaggle"></a>
- **Google Cloud** Deep Learning VM.
- **Amazon** Deep Learning AMI.
- **Docker Image**: <a href="https://hub.docker.com/r/ultralytics/ultralytics"><img src="https://img.shields.io/docker/pulls/ultralytics/ultralytics?logo=docker" alt="Docker Pulls"></a>

## Status

<a href="https://github.com/ultralytics/ultralytics/actions/workflows/ci.yaml?query=event%3Aschedule"><img src="https://github.com/ultralytics/ultralytics/actions/workflows/ci.yaml/badge.svg" alt="Ultralytics CI"></a>

If this badge is green, it means all [Ultralytics CI](https://github.com/ultralytics/ultralytics/actions/workflows/ci.yaml?query=event%3Aschedule) tests are passing, verifying the correct operation of all YOLO modes and tasks.

**Comment by Y-T-G** [comment]
The confusion matrix is not image level.

https://github.com/ultralytics/ultralytics/issues/16695#issuecomment-2394795256

**Comment by shrutichakraborty** [comment]
> The confusion matrix is not image level.
> 
> [#16695 (comment)](https://github.com/ultralytics/ultralytics/issues/16695#issuecomment-2394795256)

Hello, yes I understand that. Here is the confsuion matrix I obtain, all instances of background are falsely detected as an object. Therefore I am looking to modify the weight attributed to them while computing loss. Can you tell me more on this?

**Comment by shrutichakraborty** [comment]
> @shrutichakraborty That's how it will look. There's no such thing as true background for object detection. That last cell will always be empty.

I find that a bit confusing, for images that contain no detections they should be seen as "background" right and therefore in the True Negative cell should have some values

**Comment by Y-T-G** [comment]
No, because the confusion matrix is not image-level like I said. It's not showing the number of images. It's showing the number of objects. And there's no such thing as a background object. Images with objects also contain background, not just images with no objects. So it would be inconsistent to switch from object-level counting to image-level counting when it comes to background class.

**Comment by hashdashd** [comment]
>Therefore I am looking to modify the weight attributed to them while computing loss.

Still a valid question we would like the answer to as well. 

How do you implement class weighting during training in yolov11 for unbalanced datasets?

**Comment by Y-T-G** [comment]
@hashdashd Ultralytics doesn't support reweighting classes.

**Comment by Y-T-G** [comment]
Duplicate of https://github.com/ultralytics/ultralytics/issues/21082
